File: evidence/generated-systems/angular/insurance-on-angular/src/app/models/Document.js

// Define collection and schema for Document
export interface Document {
    fileName:
	type : string
    uploadedDate:
	type : Date
    Policy:
	type : Schema.Types.ObjectId
    Claim:
	type : Schema.Types.ObjectId
    Application:
	type : Schema.Types.ObjectId
    Customer:
	type : Schema.Types.ObjectId
    DocumentType:
 	type : String
#
    collection: 'documents'
}

File: evidence/generated-systems/golang/insurance-on-golang/insurance-on-golang/internal/model/Document.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// Document Declaration
//==============================================================
type Document struct {
    gorm.Model
     FileName                                    string
    UploadedDate                                                            time.Time
    PolicyId         *uint
    Policy           *Policy `gorm:"foreignKey:PolicyId"`
    ClaimId         *uint
    Claim           *Claim `gorm:"foreignKey:ClaimId"`
    ApplicationId         *uint
    Application           *Application `gorm:"foreignKey:ApplicationId"`
    CustomerId         *uint
    Customer           *Customer `gorm:"foreignKey:CustomerId"`
    DocumentType                      DocumentType

// parent associations as their child

}

